ggml/src/vulkan-shaders/vulkan-shaders-gen.cpp

```diff
@@ -197,7 +193,11 @@ void string_to_spv(const std::string& _name, const std::string& in_fname, const
     std::string out_fname = join_paths(output_dir, name + ".spv");
     std::string in_path = join_paths(input_dir, in_fname);
 
-    std::vector<std::string> cmd = {GLSLC, "-fshader-stage=compute", "--target-env=vulkan1.2", "-O", in_path, "-o", out_fname};
+    #ifdef _WIN32
```

Comment:
For compatibility with Windows x86, x64 and arm64, should it not better be "#if defined(_WIN32) || defined(_WIN64) || defined(_M_ARM64)"?

Reply:
According to the [documentation](https://learn.microsoft.com/en-us/cpp/preprocessor/predefined-macros), `_WIN32` should be enough:

> `_WIN32` Defined as 1 when the compilation target is 32-bit ARM, 64-bit ARM, x86, or x64. Otherwise, undefined.

Reply:
@ all - its embarrassing, I'm totally sorry, @0cc4m is right. Idiot me did not find this documentation and I wrote a stupid/wrong test program (Ok, it at least worked correctly for detecting clang vs. MSVC on WoA).